daft/unity_catalog/unity_catalog.py

```
@@ -119,9 +119,9 @@ def load_table(
                 )
         except unitycatalog.NotFoundError:
             if not new_table_storage_path:
-                raise ValueError(
-                    f"Table {table_name} is not an existing table. If a new table needs to be created, provide 'new_table_storage_path' value."
-                )
+                table_type = "MANAGED"
```

Were you able to verify this? My only concern is about silent failures. If the customer has a typo, I'd rather it error that there were no table rather than creating a new table at the typo location. I will chat with the team with this API's behavior before merging this.

---

For sure would like to split this functionality. Since it works already, how about we update the PR to accept the table type and NOT create if the table_type is MANAGED?

For future improvements, yes there's a lot of cleanup to do, and ideally I want to migrate as much functionality as possible to https://github.com/Eventual-Inc/Daft/blob/main/daft/catalog/__unity.py

---

Ok, I misunderstood this! I think there's a lot of work to improve this class, and I want to migrate functionality to the new catalog API. We have several issues with this older implementation, but let's see if we can get the minimum to close #4106 

* If the managed table already exists, then the `load_table` method should work
* If the managed table does NOT exist, then we are unable to create it with the current code.

How about we make the goal to be "creating managed unity catalog tables"? Is this accurate?

---

Exactly this was my initial goal. I would also prefer to keep it sample. Give me 20 min to make a draft change and we check it again.

---

For sure, no rush. Let me know if I can help 👍

---

@rchowell Have a look again :) I added a warning for the case if choosing table_type MANAGED. Also I put the default as EXTERNAL so people would only on purpose do it.

---

Once the logic is fine I will properly test it.

---

Logic looks good. Let me know once you've tested then we can 🚢

---

Awesome. Will do tomorrow :)